packages/viz/src/components/LineChart/LineChart.tsx

```diff
@@ -0,0 +1,404 @@
+     * If no value is provided, the chart will automatically set a min value in order for all values to be equally distributed.
+     * Set this property to `min` to use the maximum data value.
+     */
+    minValue?:
+      | string
+      | number
+      | "min"
+      | ((obj: {
+          max: string | number;
+          min: string | number;
+        }) => string | number);
+  };
+  /** Columns to use to group your measures. */
+  series?: string | string[];
+  /** Tooltip options. */
+  tooltip?: {
+    /** Whether to show the tooltip or not. Defaults to `true`. */
+    show?: boolean;
+    /** Formatter for the value in the tooltip. */
+    valueFormatter?: (value?: string | number) => string;
+  };
+  /** Legend options. */
+  legend?: {
+    /**
+     * Whether to show the legend or not.
+     *
+     * If they are multiple series, the legend will appear by default. Otherwise, the legend will not be shown.
+     */
+    showLegend?: boolean;
+  };
+  /** Formatter the lines names used on the tooltips and legend. */
+  lineNameFormatter?: (value?: string) => string;
+  /** Strategy to use when there are empty cells. Defaults to `void`. */
+  emptyCellMode?: HvChartEmptyCellMode;
+  /** Whether the area under the lines should be filled to have an area. Defaults to `false`. */
+  area?: boolean;
+  /** Whether the chart is stacked. Defaults to `false`. */
+  stacked?: boolean;
+  /** Ranger slider options for the horizontal axis. */
+  horizontalRangeSlider?: {
+    /** Whether to show the ranger slider or not. Defaults to `false`. */
+    show?: boolean;
+  };
+}
+
+/**
+ * A line chart or line plot or line graph is a type of chart which displays information as a series of data points
+ * connected by straight line segments. It is a basic type of chart common in many fields.
+ */
+export const HvLineChart = ({
+  data,
+  series,
+  xAxis,
+  measures,
+  legend,
+  tooltip,
+  lineNameFormatter,
+  area = false,
+  stacked = false,
+  emptyCellMode = "void",
+  horizontalRangeSlider,
+}: HvLineChartProps) => {
+  const { activeTheme, selectedMode, selectedTheme } = useTheme();
+  const { theme } = useVizTheme();
+
+  const chartData = useMemo(() => {
+    const tableData =
+      data.type === "row" ? from(data.values) : table(data.values);
+
+    const agFunction: HvChartAggregation | HvChartAggregation[] =
+      measures.aggregation ?? "sum";
+
+    const aggregations =
+      typeof measures.fields === "string"
+        ? {
+            [measures.fields]: getAggregation(
+              Array.isArray(agFunction) ? agFunction[0] : agFunction,
+              measures.fields
+            ),
+          }
+        : {};
+
+    if (Array.isArray(measures.fields)) {
+      for (let i = 0; i < measures.fields.length; i++) {
+        const field = measures.fields[i];
+        aggregations[field] = getAggregation(
+          Array.isArray(agFunction) ? agFunction[i] ?? "sum" : agFunction,
+          field
+        );
+      }
+    }
```

Comment:
Had some trouble reading this... It is probably easier to first ensure you always working with arrays.

```suggestion
    const measuresFields = Array.isArray(measures.fields)
      ? measures.fields
      : [measures.fields];
    const agFunction = Array.isArray(measures.aggregation)
      ? measures.aggregation
      : [measures.aggregation];

    const aggregations = measuresFields.reduce(
      (acc, field, i) => ({
        ...acc,
        [field]: getAggregation(agFunction[i] ?? "sum", field),
      }),
      {}
    );
```

Reply:
Okay, I will update this code to improve readability